custom/icds/data_management/tasks.py

```diff
@@ -1,11 +1,36 @@
+import io
+
+from django.template.loader import render_to_string
+
 from celery.task import task
 
+from corehq.util.log import send_HTML_email
 from custom.icds.data_management.models import DataManagementRequest
+from custom.icds.data_management.serializers import (
+    DataManagementRequestSerializer,
+)
 
 
 @task(queue='background_queue')
-def execute_data_management(slug, domain, db_alias, initiated_by, start_date, end_date):
-    processed, skipped = DataManagementRequest(
-        slug=slug, domain=domain, db_alias=db_alias, initiated_by=initiated_by,
-        start_date=start_date, end_date=end_date
-    ).execute()
+def execute_data_management(request_id):
+    request = DataManagementRequest.objects.get(pk=request_id)
+    processed, skipped, logs = request.execute()
+    serialized = DataManagementRequestSerializer().to_representation(request)
+    file_attachments = []
+    for filename, filepath in logs.items():
+        with open(filepath) as logfile:
+            file_attachments.append({
+                'title': filename,
+                'file_obj': io.StringIO(logfile.read()),
```

Comment:
how big is this likely to be? Could it cause memory issues?

Reply:
roughly 4MBs from last time when we ran for cases updated within last 6 months for invalid phone number fixes.
It's suppose to be run quarterly so should be that or less, though it can depend on the number of cases found faulty. The file is just a list of case ids so even for really high number of cases should not cause memory issues.